core/src/main/java/org/infinispan/configuration/parsing/CacheParser.java

```diff
@@ -1106,14 +1110,25 @@ private void parseGroups(final ConfigurationReader reader, final ConfigurationBu
    }
 
    private ConfigurationBuilder getConfigurationBuilder(ConfigurationReader reader, ConfigurationBuilderHolder holder, String name, boolean template, String baseConfigurationName) {
-      if (!reader.getProperties().containsKey(IGNORE_DUPLICATES) && holder.getNamedConfigurationBuilders().containsKey(name)) {
-         throw CONFIG.duplicateCacheName(name);
-      }
+      checkDuplicateCacheName(reader, holder, name);
       ConfigurationBuilder builder = holder.newConfigurationBuilder(name);
       builder.configuration(baseConfigurationName).template(template);
       return builder;
    }
 
+   private void checkDuplicateCacheName(ConfigurationReader reader, ConfigurationBuilderHolder holder, String name) {
+      Properties props = reader.getProperties();
+      if (props.containsKey(IGNORE_DUPLICATES) || !holder.getNamedConfigurationBuilders().containsKey(name))
+         return;
+
+      if (props.containsKey(ALLOWED_DUPLICATES)) {
+         for (String allowedDuplicate : props.getProperty(ALLOWED_DUPLICATES).split(","))
```

Comment:
Property values can be non-strings. I suggest you store them as a HashSet in `Server.java` so that here it's just a matter of `.contains()`

Reply:
Good point. Updated.